src/Fields/interpolate.jl

```diff
@@ -1,19 +1,41 @@
 using Oceananigans.Grids: XRegRectilinearGrid, YRegRectilinearGrid, ZRegRectilinearGrid
-using CUDA: allowscalar 
+using Oceananigans.Grids: XRegLatLonGrid, YRegLatLonGrid, ZRegLatLonGrid
+using Oceananigans.Grids: xnodes, ynodes, znodes, topology
+
+@inline middle_point(l, h) = Base.unsafe_trunc(Int, (l + h) / 2)
+
+@inline function index_binary_search(vec, val, array_size)
```

Comment:
This function looks like it was copied from somewhere, in which case it's best to cite the source.

Reply:
A docstring is probably appropriate too

Reply:
done!